app/assets/javascripts/search.js

```diff
@@ -0,0 +1,20 @@
+var $main        = $('#home_query');
+var $name        = $('input#name');
+var $summary     = $('input#summary');
+var $description = $('input#description');
+var $downloads   = $('input#downloads');
+var $updated     = $('input#updated');
+
+$name.add($summary)
+  .add($description)
+  .add($downloads)
+  .add($updated)
+  .on('input', function(e) {
+    var name        = $name.val().length > 0 ? 'name: ' + $name.val() : '';
+    var summary     = $summary.val().length > 0 ? 'summary: ' + $summary.val() : '';
+    var description = $description.val().length > 0 ? 'description: ' + $description.val() : '';
+    var downloads   = $downloads.val().length > 0 ? 'downloads: ' + $downloads.val() : '';
+    var updated     = $updated.val().length > 0 ? 'summary: ' + $updated.val() : '';
+
+    $main.val($.trim(name + ' ' + summary + ' ' + description + ' ' + downloads + ' ' + updated));
+});
```

Comment:
All this javascript should only be run on the advanced search page. So check for the search class or something.